vms/platformvm/network/network.go

```diff
@@ -109,60 +107,80 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
 		)
 		return nil
 	}
-
 	txID := tx.ID()
 
 	// We need to grab the context lock here to avoid racy behavior with
 	// transaction verification + mempool modifications.
+	//
+	// Invariant: tx should not be referenced again without the context lock
+	// held to avoid any data races.
 	n.ctx.Lock.Lock()
-	defer n.ctx.Lock.Unlock()
-
 	if reason := n.mempool.GetDropReason(txID); reason != nil {
 		// If the tx is being dropped - just ignore it
 		return nil
 	}
-
-	// add to mempool
-	if err := n.IssueTx(ctx, tx); err != nil {
-		n.ctx.Log.Debug("tx failed verification",
-			zap.Stringer("nodeID", nodeID),
-			zap.Error(err),
-		)
+	err = n.issueTx(tx)
+	n.ctx.Lock.Unlock()
```

Comment:
if we don't use `defer n.ctx.Lock.Unlock()` we must unlock in every code path. Add n.ctx.Lock.Unlock() within if above, line 119

Reply:
This didn't get reverted correctly, fixed